javamds/mdsobjects.c

```diff
@@ -1844,6 +1844,7 @@ JNIEXPORT void JNICALL Java_MDSplus_Tree_closeTree
   ctx = getCtx(ctx1, ctx2);
   status = _TreeClose(&ctx, (char *)name, shot);
   (*env)->ReleaseStringUTFChars(env, jname, name);
+  TreeFreeDbid(ctx);
```

Comment:
dont you have to report the change back to java ... should you not reset ctx1 and ctx2 to 0

Reply:
There is a bool flag in java that report the open/close state.